### ISSUE TITLE
Recording stop loading certain Network requests

### ISSUE THREAD

**Issue by bvaughn** [RecordReplay/devtools#6511]
These requests were loading prior to 20 minutes ago or so, then some of them stopped. Not sure why.

Loom: https://www.loom.com/share/9eadf59ecd71497bafa27df1089c888a

Replay: https://app.replay.io/recording/replay-network-request-details-not-loading--fa66a20b-1eb6-48ee-b9c7-3cb1c0b6c46e

**Comment by bvaughn** [comment]
It seems like when this is happening, `loadedRegions` is undefined. Is that expected?

**Comment by jcmorrow** [comment]
@bvaughn is that coming from Replay or did you see that in the browser JS console?

**Comment by bvaughn** [comment]
@jcmorrow I see it in my locally running Replay. I can see the no-details line being hit in my session too but reasoning about what's happening there is a bit harder due to source maps. Still looking.

**Comment by jcmorrow** [comment]
well, just beware, I made this: https://github.com/RecordReplay/devtools/issues/6512 because I don't believe that `loadedRegions` should be null when this code path gets executed. That just makes no sense to me.

**Comment by bvaughn** [comment]
Do you suspect the Apollo hooks for this? I got side tracked digging into the comments issue b'c of these paper cuts but I think I was seeing some confusing things in terms of what the cache should have (response returned from server after mutation and get) and what was actually being passed to React during render.

**Comment by jcmorrow** [comment]
Sorry I didn't see this comment til now. So, I'm sure you know that Apollo sits between the client and the GraphQL API. So the stuff that goes over that is stuff like comments, basic recording data (id's titles, which workspace it's attached to, etc.), some information about users and orgs, billing info, stuff like that. Everything that goes over the protocol, information about sources, console logs, really most of what makes devtools devtools and not just another web app with users and CRUD resources goes over our Protocol websocket API, and because none of it is GraphQL, none of it touches Apollo (at least... as far as I know 🙃 ). Anyways, I too see 👻 s in Apollo with some regularity, but unless you're dealing with the Library page or comments, it's pretty unlikely that the data you are seeing ever touched Apollo.

**Comment by jcmorrow** [comment]
And, just for posterity, we talked about this in Discord, earlier, but this is *definitely* caused by the fact that we manage loaded regions and focus as two windows that have nothing to do with each other. In truth, while they are two very different pieces of state, most of the time when we want to ask "what resources should currently be available" the correct answer is "the overlap between what is currently loaded from the backend and the current focus window". We should encode that concept into redux, I think it will make our lives easier.

**Comment by bvaughn** [comment]
What you said RE: what lives in Apollo vs the backend makes sense. I was 4 or 5 bugs in yesterday when I left my comment and I think I just wasn't thinking clearly at the time 😁  Thanks for writing that out for me!

I hadn't yet dug into how we store (and schedule React updates for) our "protocol" data, but looking this morning I think that's all going into Redux (maybe in the "app" part of the Redux store)?

> most of the time when we want to ask "what resources should currently be available" the correct answer is "the overlap between what is currently loaded from the backend and the current focus window". **We should encode that concept into redux**

Are we not already doing that?
```js
ThreadFront.listenForLoadChanges(parameters => {
  store.dispatch({ type: "set_loaded_regions", parameters });
});
```

Or are you saying that we're missing a new type of selector for Network resources? For instance, we have this selector for Console messages which takes both loaded and focused regions into account:
https://github.com/RecordReplay/devtools/blob/cb7becfbea781e278628aa4ce6b28f65c191c842/src/devtools/client/webconsole/selectors/messages.ts#L35-L62

It looks like the Network equivalent (which lives in `reducers/network.ts` for some reason?) maybe just doesn't account for loaded regions yet?
https://github.com/RecordReplay/devtools/blob/cb7becfbea781e278628aa4ce6b28f65c191c842/src/ui/reducers/network.ts#L106-L113

Sorry for the long comment. 😄

**Comment by bvaughn** [comment]
Okay. Can definitely confirm (thanks to the new advanced timeline mode I learned about from Jaril the other day) that the Network request issue I was seeing seems to be caused by me clicking on a comment to jump to a request that isn't yet in a loaded region (the orange line?). Once it loads, I can view it.

Maybe the fact that it wasn't loading for a period of time yesterday corresponded to something going wrong on the backend.

So I don't think of this as a "bug" any more so much as a bad user experience.

**Comment by bvaughn** [comment]
One way to address this would be to update the Network selector to filter events that are outside of the loaded regions. That doesn't seem like the right approach though. I think "loaded" and "focused" are different in a significant way.

Filtering them out because of a focused region seems like what the user has _asked_ us to do ("let me focus only on things in this period") where as the loaded region is just a technical limitation. Maybe that's why the selector is written the way it is. (Although this logic seems to apply to console messages as well? So I'm curious if we should revisit that bit of UX.)

My intuition is that we should fix this by showing a better message to the user than "_Request details currently unavailable_".

Seems like this was a bandaid added in #5605 that had a follow up issue filed #5606, which was then subsequently closed. I think I might suggest we re-open that issue.

**Comment by jcmorrow** [comment]
Well, there is still a bug here, which is that if the backend unloads regions with network requests in them (which it can do at any time) those network requests should (I think?) disappear from the network request table, and right now they do not.

**Comment by jcmorrow** [comment]
I guess it's reasonable to also say "network requests outside the current window are unclickable and have their opacity turned down" or something like that, but it seems like we should generally treat things that are unreachable either for focus reasons or for loading reasons the same way. I'm open to push back on that though. One thing is also for certain: right now we request the body of the request from the backend even if that region is unloaded and that should definitely not happen because it will always be a protocol error.

**Comment by bvaughn** [comment]
> Well, there is still a bug here, which is that if the backend unloads regions with network requests in them (which it can do at any time) those network requests should (I think?) disappear from the network request table, and right now they do not.

I'm not sure I agree with this (from a UX perspective) but I'd love to talk and learn more about your reasoning.

**Comment by bvaughn** [comment]
Fixed via #6534
